doc/contribute/coding_guidelines/index.rst

```diff
@@ -1255,6 +1255,39 @@ Related GitHub Issues and Pull Requests are tagged with the `Inclusive Language
 .. _CAN in Automation Inclusive Language news post: https://www.can-cia.org/news/archive/view/?tx_news_pi1%5Bnews%5D=699&tx_news_pi1%5Bday%5D=6&tx_news_pi1%5Bmonth%5D=12&tx_news_pi1%5Byear%5D=2020&cHash=784e79eb438141179386cf7c29ed9438
 .. _CAN in Automation Inclusive Language: https://can-newsletter.org/canopen/categories/
 
+
+Rule A.3: Macro name collisions
+===============================
+
+Severity
+--------
+
+Required
+
+Description
+-----------
+
+Commonly used macros such as  ``MIN``, ``MAX``, ``ARRAY_SIZE`` as well as other
+similar ones, must not be modified or protected to avoid name collisions with
+other implementations. In particular, they must not be prefixed to place them in
+a Zephyr-specific namespace, nor can they be conditionally excluded from
+compilation using ``#ifndef``.
```

Comment:
another common pattern I have seen is to `#undef` and re-define, which is something that should certainly be avoided.

```suggestion
other implementations. In particular, they must not be prefixed to place them in
a Zephyr-specific namespace, re-defined using `#undef`, or conditionally
excluded from compilation using ``#ifndef``.
```